rclcpp/src/rclcpp/executors/static_executor_entities_collector.cpp

```diff
@@ -327,7 +334,7 @@ StaticExecutorEntitiesCollector::add_callback_group(
     throw std::runtime_error("Callback group was already added to executor.");
   }
   if (is_new_node) {
-    weak_nodes_to_guard_conditions_[node_ptr] = node_ptr->get_notify_guard_condition();
+    new_nodes_.push_back(node_ptr);
```

Comment:
`new_nodes_` is updated here, and read/cleared above, is there a race in this case or are these two sections synchronized already?

Reply:
Sections were not synchronized. Added a mutex in b8db146

Also the executor's guard condition was being added twice to the memory strategy, changed that in 41febe3